x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/FilterBlockBuilder.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.data;
+
+public class FilterBlockBuilder {
```

Comment:
Can this be constructed with a BlockFactory. That way the operator an inject the factory that is be to used, rather than the inbound block determining that. For now, there is really just one block factory, but that may not always be the case.